temporal-sdk/src/test/java/io/temporal/internal/common/ProtobufTimeUtilsTest.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (C) 2022 Temporal Technologies, Inc. All Rights Reserved.
+ *
+ * Copyright (C) 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this material except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.temporal.internal.common;
+
+import static io.temporal.internal.common.ProtobufTimeUtils.MAX_SECONDS;
+import static io.temporal.internal.common.ProtobufTimeUtils.MIN_SECONDS;
+import static org.junit.Assert.assertEquals;
+
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.Collection;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameters;
+
+@RunWith(Parameterized.class)
+public class ProtobufTimeUtilsTest {
+
+  @Parameters
+  public static Collection<Object[]> data() {
```

Comment:
Can we confirm all the tests that didn't throw have the same behaviour before and after your change?

Reply:
Yes, I specifically tested that in the first commit after the rebase + force-push.

Reply:
Happily, the belief that `Durations.toMillis` does round to nearest instead of truncation was mistaken, as that would have meant that there was an asymmetry in the conversion.  Both `Duration.toMillis` (Java) and `Durations.toMillis` (protobuf) truncate.